### ISSUE TITLE
Cloudflare adapter: confusing semantics for accessing `$env/*`

### ISSUE THREAD

**Issue by cdcarson** [sveltejs/kit#9336]
### Describe the bug

**tl;dr** Using the Cloudflare adapter, are you supposed to always access environment variables via `event.platform.env`? Or can you still do it the regular way: `import { FOO } from '$env/static/private'`?

-----
Not sure if this is a bug, a feature request, a documentation request, me doing it wrong, or just the way it has to be. Sorry.

I'm using Cloudflare. The environment variables are a mix of "normal" environment variables (database urls, api keys, and so forth) and Cloudflare features, like KV namespaces...

```ts
// app.d.ts
declare global {
  namespace App {
    interface Platform {
      env: {
        FOO: string;
        BAR: import('@cloudflare/workers-types').KVNamespace;
      };
    }
  }
}
```

 In dev, miniflare populates `event.platform`...

```ts
// hooks.server.ts
import { dev } from '$app/environment';
import type { Handle } from '@sveltejs/kit';
export const handle: Handle = async ({ event, resolve }) => {
  if (dev) {
    const { devMiniflare } = await import('./miniflare');
    event.platform = await devMiniflare(event.platform);
  }
  return await resolve(event);
};
```

Miniflare uses `.dev.vars` for the "normal" variables and `wrangler.toml` for the KV bindings and so forth. 

Here's what's confusing me.  At some point I also had a `.env` file with the same "normal" variables as in `.dev.vars`. I blithely deleted `.env`, assuming that SvelteKit or the adapter was using the type in `app.d.ts` to produce `$env/*`. That's not the case. 

- The `import { FOO } from '$env/static/private'` method only worked in dev because of `.env`. 
- For some reason I was always accessing the KV namespace (`BAR` in my example) via `event.platform.env`, so the issue of it not appearing in `.env` never came up.
- AFAICT, in a deployed app, `import { FOO } from '$env/static/private'` still works. But don't quote me on that.

So, should I just entirely ignore `$env/*` and always access everything from the event? It's a minor inconvenience to pass the event around, but whatever.

If not, how should the dev environment be set up so the import statements work?  Can `$env/*` get produced statically from `app.d.ts` rather than from the runtime, at least in these bespoke platform cases? (I'm not sure this is actually a good idea, just throwing it out there.)

Thanks!

### Reproduction

Minimal setup...

https://github.com/cdcarson/sveltekit-kv-namespaces-repro 

(I committed `.env` and `.dev.vars` on purpose. Don't do this at home.)

### System Info

```Shell
System:
    OS: macOS 13.2.1
    CPU: (8) arm64 Apple M1
    Memory: 37.31 MB / 8.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.13.0 - ~/.nvm/versions/node/v18.13.0/bin/node
    npm: 8.19.3 - ~/.nvm/versions/node/v18.13.0/bin/npm
  Browsers:
    Chrome: 110.0.5481.177
    Safari: 16.3
  npmPackages:
    @sveltejs/adapter-auto: ^2.0.0 => 2.0.0 
    @sveltejs/adapter-cloudflare: ^2.2.0 => 2.2.0 
    @sveltejs/kit: ^1.5.0 => 1.10.0 
    svelte: ^3.54.0 => 3.55.1 
    vite: ^4.0.0 => 4.1.4
```

### Severity

annoyance

**Comment by Rich-Harris** [comment]
Honestly the whole `env.MY_NAMESPACE` thing (as opposed to, say, `context.kv.MY_NAMESPACE`) is something I've always found rather strange.

Anything in `$env/static/[private|public]` is populated at build time (from `.env` and/or `process.env`, i.e. it will respect any env vars you have configured in the Cloudflare Pages dashboard) and statically replaced — it doesn't depend on `env` in any way, so is guaranteed to work. This is the recommended way to use `FOO` since it will result in dead code elimination etc, unless for whatever reason you need to reuse the same worker with different env vars (not sure if that's even possible on Cloudflare Pages?).

`$env/dynamic/[private|public]` are populated from `env`:

https://github.com/sveltejs/kit/blob/5dada256dc05674a18946f8f19db1f25ba91aeaf/packages/adapter-cloudflare/src/worker.js#L9-L11

This _does_ result in a discrepancy between dev/preview and prod behaviour owing to the aforementioned strangeness — at dev/preview time, the SvelteKit-provided `env` objects...

```js
import { env as privateEnv } from '$env/dynamic/private';
import { env as publicEnv } from '$env/dynamic/public';
```

...will be populated from `.env` and `process.env`, but in production they will also include KV namespaces. 

I would recommend only using `event.platform.env` for KV namespaces, and preferring `$env/static/*` over `$env/dynamic/*` for everything else.
